lib/sles4sap/microsoft_sdaf.pm

```diff
@@ -0,0 +1,177 @@
+# SUSE's openQA tests
+#
+# Copyright SUSE LLC
+# SPDX-License-Identifier: FSFAP
+# Maintainer: QE-SAP <qe-sap@suse.de>
+#
+# Library used for Microsoft SDAF deployment
+
+package sles4sap::microsoft_sdaf;
+
+use strict;
+use warnings;
+use testapi;
+use Exporter qw(import);
+use Carp qw(croak);
+use mmapi qw(get_current_job_id);
+use utils qw(write_sut_file file_content_replace);
+use qesapdeployment qw(qesap_az_get_resource_group);
+use File::Basename;
+use Regexp::Common qw(net);
+
+=head1 SYNOPSIS
+
+Library with common functions for Microsoft SDAF deployment automation:
+https://learn.microsoft.com/en-us/azure/sap/automation/get-started
+
+=cut
+
+our @EXPORT = qw(
+  az_login
+  sdaf_prepare_ssh_keys
+  sdaf_get_deployer_ip
+);
+
+=head2 az_login
+
+ az_login();
+
+Azure login using SPN credentials defined by secret OpenQA parameters:
+
+B<_SECRET_AZURE_SPN_APPLICATION_ID>
+
+B<_SECRET_AZURE_SPN_APP_PASSWORD>
+
+B<_SECRET_AZURE_ARM_TENANT_ID>
+
+Returns 'subscription ID' on success.
+
+=cut
+
+sub az_login {
+    # Note: For login I cannot use standard PC library, because it gives credentials which are missing permissions
+    my $app_id = get_required_var('_SECRET_AZURE_SDAF_APP_ID');
+    my $app_secret = get_required_var('_SECRET_AZURE_SDAF_APP_PASSWORD');
+    my $tenant_id = get_required_var('_SECRET_AZURE_SDAF_TENANT_ID');
+    my @secret_variables = (
+        "export ARM_CLIENT_ID=$app_id",
+        "export ARM_CLIENT_SECRET=$app_secret",
+        "export ARM_TENANT_ID=$tenant_id"
+    );
+
+    write_bashrc_variables(@secret_variables);
+
+    my $login_cmd = 'while ! az login --service-principal -u ${ARM_CLIENT_ID} -p ${ARM_CLIENT_SECRET} -t ${ARM_TENANT_ID}; do sleep 10; done';
+    assert_script_run($login_cmd, timeout => 120);
+
+    my $subscription_id = script_output('az account show -o tsv --query id');
+    return ($subscription_id);
+}
+
+=head2 write_bashrc_variables
+
+    write_bashrc_variables(@variable list);
+
+=cut
+
+sub write_bashrc_variables {
+    my (@variables) = @_;
+
+    write_sut_file('/root/az_variables', join("\n", "\n", @variables));
```

Comment:
Why aren't you writing directly to .bashrc? Does this really need to be a function?

Reply:
This is to protect sensitive variables being exposed in the serial console. Simple echo just shows in the logs.